test/e2e/certificates_test.go

```diff
@@ -0,0 +1,87 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/retry"
+)
+
+// testCertificates tests the CA certificates for Windows nodes
+// The initial Kubelet CA certificate is valid for 1 year, from the date of the cluster installation.
+// Usually, the first rotation of the kubelet CA certificate is generated by the [API Server Operator](https://github.com/openshift/cluster-kube-apiserver-operator)
+// at 80% for the CA certificate lifecycle (1 year), which is 292 days, and removes the previous CA certificate after 365 days.
+func testCertificates(t *testing.T) {
+	// init test context
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	t.Run("Kubelet CA rotation", tc.testKubeletCARotation)
+}
+
+// testKubeletCARotation test the rotation of the Kubelet CA certificate. This test accounts for the certificate
+// rotation that takes place at 80% for the CA certificate lifecycle.
+// Note that the old certificate is still present in the cluster until day 365, so clients may be able to use it.
+func (tc *testContext) testKubeletCARotation(t *testing.T) {
+	// node logs must be available before rotation
+	err := checkJournalLogInWindowsNodes()
+	require.NoError(t, err, "error checking journal logs in Windows nodes")
+
+	// force CA rotation
+	err = tc.forceKubeApiServerCertificateRotation()
+	require.NoError(t, err)
+
+	// wait 5 seconds for the new certificate and add 3 seconds per node to allow
+	// WMCO to update the kubelet-ca.crt file on each node
+	nodeCount := len(gc.allNodes())
+	time.Sleep(time.Duration(5+3*nodeCount) * time.Second)
+
+	// node logs must be available after rotation
+	err = checkJournalLogInWindowsNodes()
+	assert.NoError(t, err, "error checking journal logs in Windows nodes after kubelet CA rotation")
+}
```

Comment:
I don't understand how this test works
how does checking that the Windows event log is readable indicate that the cert rotation correctly occurred?

Reply:
I feel that this test is not directly testing the functionality being added, and could potentially lead to false positives/negatives
i'd prefer just checking the certificate file on the Windows VM to ensure that the new cert is present within it.